lib/plugins/aws/package/compile/functions.js

```diff
@@ -615,9 +618,12 @@ class AwsCompileFunctions {
           this.ensureTargetExecutionPermission(destinations.onFailure);
         }
         destinationConfig.OnFailure = {
-          Destination: destinations.onFailure.startsWith('arn:')
-            ? destinations.onFailure
-            : this.provider.resolveFunctionArn(destinations.onFailure),
+          Destination:
+            (typeof destinations.onFailure === 'string' &&
```

Comment:
this resolution logic sounds like a repetition between `onSuccess` and `onFailure` - maybe it could be abstracted to a small function that can be reused?

Reply:
You're right. I'll fix it.